metasource/driver/rand.go

```diff
@@ -0,0 +1,12 @@
+package driver
+
+import (
+	"crypto/rand"
+	"fmt"
+)
+
+func randomHex(n int) string {
```

Comment:
Also, for the sake of simplicity

```suggestion
func GenerateIdentity(n int) string {
```

Reply:
You should have the default length as a global variable inside of the configuration module.

Reply:
Utilised string pointer and used default length as a global variable.